src/Emulator/Peripherals/Peripherals/MTD/OpenTitan_FlashController.cs

```diff
@@ -0,0 +1,734 @@
+                .WithFlag(2, out defaultMpRegionEraseEnabled, name: "ERASE_EN")
+                .WithTaggedFlag("SCRAMBLE_EN", 3)
+                .WithIgnoredBits(4, 1 + 31 - 4);
+            
+            // TODO(julianmb): support register write enable. this isnt tested in the unittests currently
+            Registers.BankCfgRegwen.Define(this, 0x1)
+                .WithFlag(0, name: "BANK", mode: FieldMode.Read | FieldMode.WriteZeroToClear)
+                .WithIgnoredBits(1, 31);
+            Registers.MpBankCfg.Define(this)
+                .WithFlag(0, out eraseBank0, name: "ERASE_EN_0")
+                .WithFlag(1, out eraseBank1, name: "ERASE_EN_1")
+                .WithIgnoredBits(2, 1 + 31 - 2);
+            
+            Registers.OpStatus.Define(this)
+                .WithFlag(0, out opStatusRegisterDoneFlag, name: "done")
+                .WithFlag(1, out opStatusRegisterErrorFlag, name: "err")
+                .WithIgnoredBits(2, 1 + 31 - 2);
+            Registers.Status.Define(this, 0xa)
+                .WithFlag(0, name: "rd_full", mode: FieldMode.Read)
+                .WithFlag(1, out statusRegisterReadEmptyFlag, name: "rd_empty", mode: FieldMode.Read)
+                .WithFlag(2, name: "prog_full", mode: FieldMode.Read)
+                .WithFlag(3, name: "prog_empty", mode: FieldMode.Read)
+                .WithFlag(4, name: "init_wip", mode: FieldMode.Read)
+                .WithIgnoredBits(5, 1 + 7 - 5)
+                .WithValueField(8, 1 + 16 - 8, name: "error_addr", mode: FieldMode.Read)
+                .WithIgnoredBits(17, 1 + 31 - 17);
+            Registers.PhyStatus.Define(this, 0x6)
+                .WithFlag(0, name: "init_wip", mode: FieldMode.Read)
+                .WithFlag(1, name: "prog_normal_avail", mode: FieldMode.Read)
+                .WithFlag(2, name: "prog_repair_avail", mode: FieldMode.Read)
+                .WithIgnoredBits(3, 1 + 31 - 3);
+            
+            Registers.Scratch.Define(this)
+                .WithValueField(0, 32, name: "data");
+            Registers.FifoLevel.Define(this, 0xf0f)
+                .WithValueField(0, 5, name: "PROG")
+                .WithReservedBits(5, 1 + 7 - 5)
+                .WithValueField(8, 1 + 12 - 8, name: "RD")
+                .WithIgnoredBits(13, 1 + 31 - 13);
+
+            // TODO(julianmb): implement fifo reset. There isnt any unittest for this currently.
+            Registers.FifoRst.Define(this)
+                .WithFlag(0, name: "EN")
+                .WithIgnoredBits(1, 31);
+            // TODO(julianmb): handle writes while fifo is full. There isnt any unittest for this currently.
+            Registers.ProgramFifo.Define(this)
+                .WithValueField(0, 32, mode: FieldMode.Write, writeCallback: (_, n) => {
+                    uint controlValue = RegistersCollection.Read((long)Registers.Control);
+                    dynamic programFlash = flashSelectPartition.Value ? (dynamic)infoFlash : (dynamic)dataFlash;
```

Comment:
No need to use `dynamic` here. I suggest changing it to:
```suggestion
                    var programFlash = flashSelectPartition.Value ? infoFlash : (IDoubleWordPeripheral)dataFlash;
```
This way the check is verified at compile time.